test/ProtocolRewards.t.sol

```diff
@@ -111,12 +117,32 @@ contract ProtocolRewardsTest is Test {
         uint256 creatorRewardsBalance = protocolRewards.balanceOf(creator);
 
         vm.prank(creator);
+
+        vm.expectEmit();
+        emit Withdraw(creator, creator, creatorRewardsBalance);
+
         protocolRewards.withdraw(creator, creatorRewardsBalance);
 
         assertEq(creator.balance, beforeCreatorBalance + creatorRewardsBalance);
         assertEq(protocolRewards.totalSupply(), beforeTotalSupply - creatorRewardsBalance);
     }
 
+    function testfuzz_Withdraw(uint256 amount) public {
+        uint256 beforeCreatorBalance = creator.balance;
+        uint256 beforeTotalSupply = protocolRewards.totalSupply();
+
+        vm.assume(amount <= protocolRewards.balanceOf(creator));
```

Comment:
I have a preference for [bounds ](https://book.getfoundry.sh/reference/forge-std/bound?highlight=boun#bound) over assumes whenever possible - a `bound` transforms the input such that the `param` is within bounds where as an `assume` will _discard_ a test if it violates the assumption. Generally assume is best used for second order effects vs base level params.

Here it would look like:
```amount = bound(amount, 0,  protocolRewards.balanceOf(creator));```